apps/condo/domains/user/components/auth/ValidatePhoneForm.tsx

```diff
@@ -0,0 +1,222 @@
+    const ChangePhoneNumberLabel = intl.formatMessage({ id: 'pages.auth.register.ChangePhoneNumber' })
+    const FieldIsRequiredMsg = intl.formatMessage({ id: 'FieldIsRequired' })
+    const SmsCodeTitle = intl.formatMessage({ id: 'pages.auth.register.field.SmsCode' })
+    const ResendSmsLabel = intl.formatMessage({ id: 'pages.auth.register.ResendSmsLabel' })
+    const SMSCodeMismatchError = intl.formatMessage({ id: 'pages.auth.register.SMSCodeMismatchError' })
+    const SMSExpiredError = intl.formatMessage({ id: 'pages.auth.register.SMSExpiredError' })
+    const ConfirmActionExpiredError = intl.formatMessage({ id: 'pages.auth.register.ConfirmActionExpiredError' })
+    const SMSMaxRetriesReachedError = intl.formatMessage({ id: 'pages.auth.register.SMSMaxRetriesReachedError' })
+    const SMSBadFormat = intl.formatMessage({ id: 'pages.auth.register.SMSBadFormat' })
+    const SMSTooManyRequestsError = intl.formatMessage({ id: 'pages.auth.TooManyRequests' })
+    const ErrorToFormFieldMsgMapping = useMemo(() => {
+        return {
+            [CONFIRM_PHONE_SMS_CODE_VERIFICATION_FAILED]: {
+                name: 'smsCode',
+                errors: [SMSCodeMismatchError],
+            },
+            [CONFIRM_PHONE_ACTION_EXPIRED]: {
+                name: 'smsCode',
+                errors: [ConfirmActionExpiredError],
+            },
+            [CONFIRM_PHONE_SMS_CODE_EXPIRED]: {
+                name: 'smsCode',
+                errors: [SMSExpiredError],
+            },
+            [CONFIRM_PHONE_SMS_CODE_MAX_RETRIES_REACHED]: {
+                name: 'smsCode',
+                errors: [SMSMaxRetriesReachedError],
+            },
+            [TOO_MANY_REQUESTS]: {
+                name: 'smsCode',
+                errors: [SMSTooManyRequestsError],
+            },
+        }
+    }, [intl])
+
+    const [form] = Form.useForm()
+    const { token, phone, handleReCaptchaVerify } = useContext(RegisterContext)
+    const [showPhone, setShowPhone] = useState(phone)
+    const [isPhoneVisible, setIsPhoneVisible] = useState(false)
+    const [phoneValidateError, setPhoneValidateError] = useState(null)
+    const [resendSmsMutation] = useMutation(RESEND_CONFIRM_PHONE_SMS_MUTATION)
+    const [completeConfirmPhoneMutation] = useMutation(COMPLETE_CONFIRM_PHONE_MUTATION)
+
+    const PhoneToggleLabel = isPhoneVisible ? intl.formatMessage({ id: 'Hide' }) : intl.formatMessage({ id: 'Show' })
+
+
+    const resendSms = useCallback(async () => {
+        const sender = getClientSideSenderInfo()
+        const captcha = await handleReCaptchaVerify('resend_sms')
+        const variables = { data: { token, sender, captcha } }
+        // @ts-ignore TODO(Dimitreee): remove after runMutation typo
+        return runMutation({
+            mutation: resendSmsMutation,
+            variables,
+            intl,
+            form,
+            ErrorToFormFieldMsgMapping,
+        }).catch(error => {
+            console.error(error)
```

Comment:
May be add notification to user?

Reply:
@pahaz i'm not sure i knew how exactly and what about notify user, guess @dkoviazin can help's here.